godot-core/src/docs.rs

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+macro_rules! define {
+    (struct $x:ident { $($f:ident),+ }) => {
+        #[derive(Clone, Copy, Debug, Default)]
+        pub struct $x {
+            $(pub $f: &'static str,)+
+        }
+    }
+}
+
+define!(struct DefinitionDocs { base, description, members });
+define!(struct ImplementationDocs { methods, signals, constants });
```

Comment:
This macro is not necessary, it barely saves any code and makes things harder to understand.